build/cmake/init.cmake

```diff
@@ -36,6 +36,26 @@ if(MSVC)
               "Flags used by the CXX compiler during ${cfg_upper} builds." FORCE)
         endif()
     endforeach()
+    
+    if(wxUSE_OPTIMISE)
+        set(MVC_PREFERRED_RELEASE_FLAGS " /LTCG /OPT:REF,ICF /DEBUG")
+        wx_string_append(CMAKE_EXE_LINKER_FLAGS_RELEASE "${MVC_PREFERRED_RELEASE_FLAGS}")
+        wx_string_append(CMAKE_SHARED_LINKER_FLAGS_RELEASE "${MVC_PREFERRED_RELEASE_FLAGS}")
+        wx_string_append(CMAKE_CXX_FLAGS_RELEASE " -DNDEBUG /Ox /Ob2 /Oi /Ot /Oy /GS- /Gy /GR- /GL /Gw")
+    endif()
+    
+    set(MSC_PDB_FLAG " /DEBUG") 
+    wx_string_append(CMAKE_EXE_LINKER_FLAGS_RELEASE "${MSC_PDB_FLAG}")
+    wx_string_append(CMAKE_SHARED_LINKER_FLAGS_RELEASE "${MSC_PDB_FLAG}")
+elseif(CMAKE_COMPILER_IS_GNUCC)
+    if(wxUSE_OPTIMISE)
+        set(GCC_STRIPPING_FLAG " -g") 
+        set(GCC_PREFERRED_RELEASE_FLAGS "-O2 -DNDEBUG -fomit-frame-pointer ${GCC_STRIPPING_FLAG}")
+        wx_string_append(CMAKE_CXX_FLAGS_RELEASE "${GCC_PREFERRED_RELEASE_FLAGS}")
+        wx_string_append(CMAKE_C_FLAGS_RELEASE "${GCC_PREFERRED_RELEASE_FLAGS}")
+        wx_string_append(CMAKE_EXE_LINKER_FLAGS_RELEASE "${GCC_STRIPPING_FLAG}")
+        wx_string_append(CMAKE_SHARED_LINKER_FLAGS_RELEASE "${GCC_STRIPPING_FLAG}")
```

Comment:
Maybe `CMAKE_STATIC_LINKER_FLAGS_RELEASE` as well?

Reply:
AFAIK but there is no linking if you are creating static library, just archiving. If I'm not right, please correct me.

Reply:
Just noticed you need to use `-s` instead of `-g`. And you're right that it is probably only used by ld.

Reply:
_-s_ means stripping, is it good idea if we plan to analyze coredumps?

Reply:
I don't know enough about gcc/ld or the `-s` flag to comment on this.
But I was referring to `-g` typo here: `set(GCC_STRIPPING_FLAG " -g")`

Reply:
https://stackoverflow.com/questions/2251391/core-dump-in-linux - it would be better to build executables with _-g_ option and strip symbols before contribution

Reply:
https://trac.wxwidgets.org/ticket/18436

Reply:
wxUSE_STRIPPED_RELEASE was added to resolve this question